--- src/lib/app/qzcommon.cpp	2015-02-13 14:14:06.392538509 -0800
+++ src/lib/app/qzcommon-fix.cpp	2015-02-13 14:20:01.269646371 -0800
@@ -25,7 +25,7 @@
 
 QUPZILLA_EXPORT const char* APPNAME = "QupZilla";
 QUPZILLA_EXPORT const char* VERSION = QUPZILLA_VERSION;
-QUPZILLA_EXPORT const char* BUILDTIME =  __DATE__" "__TIME__;
+QUPZILLA_EXPORT const char* BUILDTIME =  __DATE__ " " __TIME__;
 QUPZILLA_EXPORT const char* AUTHOR = "David Rosca";
 QUPZILLA_EXPORT const char* COPYRIGHT = "2010-2014";
 QUPZILLA_EXPORT const char* WWWADDRESS = "http://www.qupzilla.com";
