diff --git a/pcmanfm/desktopwindow.cpp b/pcmanfm/desktopwindow.cpp
index 8e90c66..b7d091c 100644
--- a/pcmanfm/desktopwindow.cpp
+++ b/pcmanfm/desktopwindow.cpp
@@ -198,6 +198,8 @@ void DesktopWindow::resizeEvent(QResizeEvent* event) {
 }

 void DesktopWindow::setDesktopFolder() {
+  if(!proxyModel_)
+    return;
   model_ = Fm::CachedFolderModel::modelFromPath(fm_path_get_desktop());
   proxyModel_->setSourceModel(model_);
 }
@@ -483,6 +485,8 @@ void DesktopWindow::onIndexesMoved(const QModelIndexList& indexes) {
 // QListView does item layout in a very inflexible way, so let's do our custom layout again.
 // FIXME: this is very inefficient, but due to the design flaw of QListView, this is currently the only workaround.
 void DesktopWindow::relayoutItems() {
+  if (!proxyModel_)
+    return;
   // qDebug("relayoutItems()");
   if(relayoutTimer_) {
     // this slot might be called from the timer, so we cannot delete it directly here.
