diff --git a/libfm-qt/foldermenu.cpp b/libfm-qt/foldermenu.cpp
index 692f114..2b8c24b 100644
--- a/libfm-qt/foldermenu.cpp
+++ b/libfm-qt/foldermenu.cpp
@@ -57,6 +57,7 @@ FolderMenu::FolderMenu(FolderView* view, QWidget* parent):

   separator3_ = addSeparator();

+  if(model) {
   sortAction_ = new QAction(tr("Sorting"), this);
   addAction(sortAction_);
   createSortMenu();
@@ -69,6 +70,7 @@ FolderMenu::FolderMenu(FolderView* view, QWidget* parent):
   connect(showHiddenAction_, &QAction::triggered, this, &FolderMenu::onShowHiddenActionTriggered);

   separator4_ = addSeparator();
+  }

   propertiesAction_ = new QAction(tr("Folder Pr&operties"), this);
   addAction(propertiesAction_);
diff --git a/libfm-qt/folderview.cpp b/libfm-qt/folderview.cpp
index d616c49..bb5576d 100644
--- a/libfm-qt/folderview.cpp
+++ b/libfm-qt/folderview.cpp
@@ -926,8 +926,6 @@ void FolderView::onFileClicked(int type, FmFileInfo* fileInfo) {
       }
     }
     else {
-      FmFolder* _folder = folder();
-      FmFileInfo* info = fm_folder_get_info(_folder);
       Fm::FolderMenu* folderMenu = new Fm::FolderMenu(this);
       prepareFolderMenu(folderMenu);
       menu = folderMenu;
